### PR TITLE
[alertmanager] Add k8s email receiver config

### PR DESCRIPTION
**Pull request by uwe-mayer**

### CODE REVIEW

**Review by bassel-z — CHANGES_REQUESTED** [review]
Thanks for the PR. Please see comments

File: global/prometheus-alertmanager-operated/templates/_alertmanager.yaml.tpl [hunk 1]
```diff
@@ -1481,3 +1485,38 @@ receivers:
           Sentry: {{"'{{template \"pagerduty.sapcc.sentry\" . }}'"}}
           Playbook: {{"'{{template \"pagerduty.sapcc.playbook\" . }}'"}}
           firing: {{"'{{ template \"pagerduty.sapcc.firing\" . }}'"}}
+
+  # email receiver config for k8s alerting
+  - name: email_k8s_alerting
+    email_configs:
+    - to: {{"'{{ .CommonLabels.primary_contact_email }}'"}}
```
[the remark this answers is not in the document]
Comment (uwe-mayer) [reply global/prometheus-alertmanager-operated/templates/_alertmanager.yaml.tpl]: fixed by a more generic approach for an e-mail receiver

File: global/prometheus-alertmanager-operated/templates/_alertmanager.yaml.tpl [hunk 2]
```diff
@@ -1481,3 +1485,38 @@ receivers:
...
+      from: {{ required ".Values.email_k8s_alerting.email_from_address undefined" .Values.email_k8s_alerting.email_from_address | quote }}
+      headers:
+        subject: {{"'{{ .CommonAnnotations.mail_subject }}'"}}
+      text: {{"'{{ .CommonAnnotations.mail_body }}'"}}
+      html: {{"'{{ .CommonAnnotations.mail_body }}'"}}
+      smarthost: {{ required ".Values.email_k8s_alerting.smtp_host undefined" .Values.email_k8s_alerting.smtp_host | quote }}
+      auth_username: {{ required ".Values.email_k8s_alerting.auth_username undefined" .Values.email_k8s_alerting.auth_username | quote }}
+      auth_password: {{ required ".Values.email_k8s_alerting.auth_password undefined" .Values.email_k8s_alerting.auth_password | quote }}
+    {{ if .Values.email_k8s_alerting.email_bcc_to_address }}
+    - to: {{ .Values.email_k8s_alerting.email_bcc_to_address }}
```
[the remark this answers is not in the document]
Comment (uwe-mayer) [reply global/prometheus-alertmanager-operated/templates/_alertmanager.yaml.tpl]: Thanks for the input, I now opted for a more generic approach to the e-mail receiver making use of the `to` envelope and the headers as suggested

**Review by bassel-z — APPROVED** [review]
LGTM

**Pull request merged**